internal/engine/eval/rego/result.go

```diff
@@ -250,15 +252,17 @@ func (srb *stringResultBuilder) addResult(msg any) error {
 	return nil
 }
 
-func (srb *stringResultBuilder) formatResults() error {
-	return engerrors.NewDetailedErrEvaluationFailed(
-		templates.RegoConstraints,
-		map[string]any{
-			"violations": srb.results,
-		},
-		"Evaluation failures: \n - %s",
-		strings.Join(srb.results, "\n - "),
-	)
+func (srb *stringResultBuilder) formatResults() (*interfaces.EvaluationResult, error) {
+	return &interfaces.EvaluationResult{
+			Output: srb.results,
+		}, engerrors.NewDetailedErrEvaluationFailed(
+			templates.RegoConstraints,
+			map[string]any{
+				"violations": srb.results,
+			},
```

Comment:
In the future, should we feed the `Output` from the first argument into the `Template` for the EvaluationError, rather than duplicating between the two?

(That should definitely be a follow-up issue)

Reply:
Yes, we should also use the `EvaluationError` for actual unexpected errors and the `EvaluationResult` for failures.